src/main/java/org/opensearch/knn/index/query/nativelib/NativeEngineKnnVectorQuery.java

```diff
@@ -83,14 +89,19 @@ public Weight createWeight(IndexSearcher indexSearcher, ScoreMode scoreMode, flo
             perLeafResults = retrieveAll(indexSearcher, leafReaderContexts, knnWeight, perLeafResults, rescoreContext == null);
             long time_in_millis = stopWatch.stop().totalTime().millis();
             if (log.isDebugEnabled()) {
-                long totalNestedDocs = perLeafResults.stream().mapToLong(perLeafResult -> perLeafResult.getResult().size()).sum();
+                long totalNestedDocs = perLeafResults.stream().mapToLong(perLeafResult -> perLeafResult.getResult().scoreDocs.length).sum();
                 log.debug("Expanding of nested docs took {} ms. totalNestedDocs:{} ", time_in_millis, totalNestedDocs);
             }
         }
 
         TopDocs[] topDocs = new TopDocs[perLeafResults.size()];
         for (int i = 0; i < perLeafResults.size(); i++) {
-            topDocs[i] = ResultUtil.resultMapToTopDocs(perLeafResults.get(i).getResult(), leafReaderContexts.get(i).docBase);
+            TopDocs leafTopDocs = perLeafResults.get(i).getResult();
+            Arrays.sort(leafTopDocs.scoreDocs, Comparator.comparing(a -> a.score));
```

Comment:
are these not already sorted? Also, is their any contract on the ordering of scoreDocs that this could cause an issue with?

Reply:
Good point. Currently it is only sorted for AnnSearch, not for exact search. I will sort in exact search as well so that we don't have to sort here.